### PR TITLE
Feature/roc 9582

### PR DESCRIPTION
**Pull request by jaberelfasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-9582

### Change description ###
As a defendant LR once I have gained access to the case raised against my defendant who has paid less than the amount claimed
I want to be able to provide the reason why my client has paid less than the total amount claimed
So that my response is captured within the system.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java [hunk 1]
```diff
@@ -96,6 +97,20 @@ private CallbackResponse handleDefendAllClaim(CallbackParams callbackParams) {
         }
         if ("DEFENDANT_RESPONSE_SPEC".equals(callbackParams.getRequest().getEventId())) {
 
+            if ("HAS_PAID_THE_AMOUNT_CLAIMED".equals(caseData.getDefenceRouteRequired())
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:100]: I thought you had created a Constants class for this

Comment (bhssamray) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:100]: Please can you write some tests to cover this logic. You may follow existing test styles around this part of the code

Comment (jaberelfasi) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:100]: this was covered in Feature/roc 9698

Comment (jaberelfasi) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:100]: can we add this bit for in a future capacity? as this was from the last sprint

**Review by bhssamray — APPROVED** [review]

**Review by sampankumar — APPROVED** [review]

**Pull request merged**